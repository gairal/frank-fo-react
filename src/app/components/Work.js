import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Grid } from '@material-ui/core';
import XpCard from './core/XpCard';

export default class Work extends Component {
  static propTypes = {
    load: PropTypes.func.isRequired,
    works: PropTypes.arrayOf(PropTypes.shape()).isRequired,
  };

  componentDidMount() {
    const { load } = this.props;
    load();
  }

  render() {
    const { works } = this.props;
    return (
      <section>
        <Grid container>
          {works.map(work => (
            <Grid item xs={12} key={work.date_in}>
              <XpCard
                img={work.image.name}
                name={work.name}
                title={work.title}
                date_in={work.date_in}
                date_out={work.date_out}
                place={work.place}
                subHeadline={work.description}
              />
            </Grid>
          ))}
        </Grid>
      </section>
    );
  }
}
