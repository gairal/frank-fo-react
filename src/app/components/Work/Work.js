import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Grid } from 'material-ui';
import XpCard from '../core/XpCard';

export default class Work extends Component {
  static propTypes = {
    load: PropTypes.func.isRequired,
    works: PropTypes.arrayOf(PropTypes.object).isRequired,
  }

  componentDidMount() {
    this.props.load();
  }

  render() {
    return (
      <section>
        <Grid container>
          {this.props.works.map(work => (
            <Grid
              item
              xs={12}
              key={work.date_in}
            >
              <XpCard
                img={work.image.name}
                name={work.name}
                title={work.title}
                date_in={work.date_in}
                date_out={work.date_out}
                place={work.place}
                subHeadline={work.description}
              />
            </Grid>
          ))}
        </Grid>
      </section>
    );
  }
}
