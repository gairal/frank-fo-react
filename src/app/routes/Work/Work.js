import AbstractRoute from '../AbstractRoute';
import component from '../../components/Work';

export default class Work extends AbstractRoute {
  constructor(store) {
    super(store, 'works', 'work');
    this.component = component;

    this.ACTIONS = {
      FETCH_SUCCESS: 'WORK_FETCH_SUCCESS',
      FETCH_FAILURE: 'WORK_FETCH_FAILURE',
    };
    this.init();
  }

  static success(json) {
    return {
      type: 'WORK_FETCH_SUCCESS',
      payload: {
        json,
      },
    };
  }

  static fail(err) {
    return {
      type: 'WORK_FETCH_FAILURE',
      payload: {
        err,
      },
    };
  }
}
