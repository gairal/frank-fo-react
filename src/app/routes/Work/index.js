import Work from './Work';

export default store => new Work(store).route;
