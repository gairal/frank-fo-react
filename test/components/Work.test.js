import React from 'react';
import renderer from 'react-test-renderer';
import Work from '@/components/Work';

describe('components > Work', () => {
  it('Renders correctly default', () => {
    const component = renderer.create(<Work
      load={jest.fn}
      works={[
        {
          name: 'test',
          title: 'test',
          place: 'test',
          date_in: '2018-01-01',
          description: 'test',
          image: { name: 'test' },
        },
      ]}
    />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
