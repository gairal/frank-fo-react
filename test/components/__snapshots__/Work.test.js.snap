// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components > Work Renders correctly default 1`] = `
<section>
  <div
    className="MuiGrid-container-1"
  >
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-12-41"
    >
      <div
        className="MuiPaper-root-99 MuiPaper-elevation4-105 MuiPaper-rounded-100 MuiCard-root-98"
        style={
          Object {
            "display": "flex",
            "marginBottom": "1rem",
            "padding": "1rem",
          }
        }
      >
        <div
          className="MuiCardMedia-root-126 card-media"
          style={
            Object {
              "backgroundImage": "url(\\"/img/test\\")",
              "flex": "none",
            }
          }
          title="test"
        />
        <div
          className="MuiCardContent-root-128"
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <h2
            className="MuiTypography-root-129 MuiTypography-title-135"
          >
            test | test
          </h2>
          <div
            className="MuiGrid-container-1 card-chips"
            style={
              Object {
                "textTransform": "uppercase",
              }
            }
          >
            <div
              className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 MuiGrid-grid-md-true-57"
              style={
                Object {
                  "flexGrow": 0,
                  "marginBottom": ".5rem",
                  "marginRight": "0.2rem",
                }
              }
            >
              <div
                className="MuiChip-root-154"
                onKeyDown={[Function]}
                role="button"
                tabIndex={-1}
              >
                <div
                  className="MuiAvatar-root-171 MuiAvatar-colorDefault-172 MuiChip-avatar-163"
                >
                  <span
                    aria-hidden="true"
                    className="material-icons MuiIcon-root-174 MuiChip-avatarChildren-166"
                    style={
                      Object {
                        "color": "#fff",
                        "fontSize": "19px",
                      }
                    }
                  >
                    date_range
                  </span>
                </div>
                <span
                  className="MuiChip-label-167"
                >
                  2017 December - present
                </span>
              </div>
            </div>
            <div
              className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 MuiGrid-grid-md-true-57"
            >
              <div
                className="MuiChip-root-154"
                onKeyDown={[Function]}
                role="button"
                tabIndex={-1}
              >
                <div
                  className="MuiAvatar-root-171 MuiAvatar-colorDefault-172 MuiChip-avatar-163"
                >
                  <span
                    aria-hidden="true"
                    className="material-icons MuiIcon-root-174 MuiChip-avatarChildren-166"
                    style={
                      Object {
                        "color": "#fff",
                        "fontSize": "19px",
                      }
                    }
                  >
                    location_city
                  </span>
                </div>
                <span
                  className="MuiChip-label-167"
                >
                  test
                </span>
              </div>
            </div>
          </div>
          <p
            className="MuiTypography-root-129 MuiTypography-caption-139"
            style={
              Object {
                "marginTop": "0.5rem",
              }
            }
          >
            test
          </p>
        </div>
      </div>
    </div>
  </div>
</section>
`;
